packages/common/src/doc-cache.ts

```diff
@@ -0,0 +1,71 @@
+import { LRUMap } from 'lru_map'
+import DocID from "@ceramicnetwork/docid"
+
+import { DocStateHolder } from "./doctype"
+
+/**
+ * Encapsulates document caching (base, commits)
+ */
+export class DocCache {
+    private readonly _cacheCommits: boolean
+    private readonly _baseCache: LRUMap<string, DocStateHolder>
+    private readonly _commitCache: LRUMap<string, DocStateHolder>
+
+    constructor(limit, cacheCommits = true) {
+        this._cacheCommits = cacheCommits
+        this._baseCache = new LRUMap(limit)
+
+        // use the same 'limit' if cacheCommits is enabled
+        this._commitCache = this._cacheCommits? new LRUMap(limit) : new LRUMap(0)
+    }
+
+    /**
+     * Sets to cache
+     * @param doc - DocStateHolder instance
+     */
+    set(doc: DocStateHolder): void {
```

Comment:
definitely a nitpick, but I'd probably call this `put` rather than `set` since it's not a KV-map setting a value to a specific key

Reply:
yeah, I wanted to call it `put` as well. The `LRUMap` uses `get` which was strange. I'll rename it